modules/3d/src/pointcloud/io_ply.cpp

```diff
@@ -386,11 +438,17 @@ void PlyDecoder::parseBody(std::ifstream &file, std::vector<Point3f> &points, st
             offset = offsetof(VertexFields, ny);
         if (p.name == "nz")
             offset = offsetof(VertexFields, nz);
-        if (p.name == "red")
+        if (p.name == "texture_u" || p.name == "s")
+            offset = offsetof(VertexFields, u);
+        if (p.name == "texture_v" || p.name == "t")
+            offset = offsetof(VertexFields, v);
+        if (p.name == "texture_w")
+            offset = offsetof(VertexFields, w);
+        if (p.name == "red"   || p.name == "diffuse_red")
             offset = offsetof(VertexFields, r);
-        if (p.name == "green")
+        if (p.name == "green" || p.name == "diffuse_green")
             offset = offsetof(VertexFields, g);
-        if (p.name == "blue")
+        if (p.name == "blue"  || p.name == "diffuse_blue")
```

Comment:
Are regular colors and diffuse colors possible together? If true, then parser should not mix red, diffuse_green and blue together, but prefer on of color spaces.

Reply:
I've never seen such PLY files; maybe we'd better add such a check.